lib/SimpleSAML/XHTML/Template.php

```diff
@@ -261,17 +264,20 @@ private function setupTwigTemplatepaths()
 
     /**
      * Setup twig.
-     * @return Twig_Environment|false
+     * @return \Twig\Environment
+     * @throws \Exception if the template does not exist
      */
     private function setupTwig()
     {
         $auto_reload = $this->configuration->getBoolean('template.auto_reload', true);
         $cache = $this->configuration->getString('template.cache', false);
+
         // set up template paths
         $loader = $this->setupTwigTemplatepaths();
+
         // abort if twig template does not exist
         if (!$loader->exists($this->twig_template)) {
-            return false;
```

Comment:
This leads to Psalm-issues all over the place because now `$this->twig` can be false and we don't test for it..
It does however not make any sense to continue execution if the template-file does not exist, so raising an exception makes much more sense. This seems like a remnant from the days where we would fall back to php-templates if no Twig-template exists.

Reply:
This makes me realize of a possible issue. Is it still possible for modules using twig templates already to work, without `usenewui` being set to `true`?

Reply:
I think not.. We completely separated it last year..

Reply:
I need to check it out, because I know some third-party modules are already implementing twig templates, and it would be unfortunate for those...